src/interface/swarm.cpp

```diff
@@ -268,80 +266,59 @@ void Swarm::setPoolMax(const int nmax_pool) {
     free_indices_.push_back(n + n_new_begin);
   }
 
-  // Resize and copy data
-  mask_.Get().Resize(nmax_pool);
-  auto mask_data = mask_.Get();
-  pmb->par_for(
-      "setPoolMax_mask", nmax_pool_, nmax_pool - 1,
-      KOKKOS_LAMBDA(const int n) { mask_data(n) = 0; });
-
-  marked_for_removal_.Get().Resize(nmax_pool);
-  auto marked_for_removal_data = marked_for_removal_.Get();
-  pmb->par_for(
-      "setPoolMax_marked_for_removal", nmax_pool_, nmax_pool - 1,
-      KOKKOS_LAMBDA(const int n) { marked_for_removal_data(n) = false; });
+  // Rely on Kokkos setting the newly added values to false for these arrays
+  Kokkos::resize(mask_, nmax_pool);
+  Kokkos::resize(marked_for_removal_, nmax_pool);
 
   Kokkos::resize(cellSorted_, nmax_pool);
 
-  neighbor_send_index_.Get().Resize(nmax_pool);
-
   blockIndex_.Resize(nmax_pool);
 
   auto &intMap_ = std::get<getType<int>()>(Maps_);
   auto &intVector_ = std::get<getType<int>()>(Vectors_);
   auto &realMap_ = std::get<getType<Real>()>(Maps_);
   auto &realVector_ = std::get<getType<Real>()>(Vectors_);
 
-  // TODO(BRR) Use ParticleVariable packs to reduce kernel launches
   for (int n = 0; n < intVector_.size(); n++) {
-    auto oldvar = intVector_[n];
-    auto newvar = std::make_shared<ParticleVariable<int>>(oldvar->label(), nmax_pool,
-                                                          oldvar->metadata());
-    auto oldvar_data = oldvar->data;
-    auto newvar_data = newvar->data;
-    pmb->par_for(
-        "setPoolMax_int", 0, nmax_pool_ - 1,
-        KOKKOS_LAMBDA(const int m) { newvar_data(m) = oldvar_data(m); });
-
-    intVector_[n] = newvar;
-    intMap_[oldvar->label()] = newvar;
+    auto &data = intVector_[n]->data;
+    std::array<int, 6> dim;
+    for (int i = 0; i < 6; i++) {
+      dim[i] = data.GetDim(6 - i);
+    }
+    data.Resize(dim[0], dim[1], dim[2], dim[3], dim[4], nmax_pool);
```

Comment:
Why not 
```c++
for (auto& data : intVector) 
     data.Resize(data.GetDim[6], data.GetDim[5], data.GetDim[4], 
                 data.GetDim[3], data.GetDim[2], nmax_pool);
```

Reply:
Yes this is much better, thanks (although I do keep curly brackets around even one-line `for` loops, as a personal eccentricity)